applications/dashboard/controllers/Api/RolesApiController.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+use Garden\Schema\Schema;
+use Garden\Web\Data;
+use Garden\Web\Exception\NotFoundException;
+use Garden\Web\Exception\ServerException;
+use Vanilla\Utility\CapitalCaseScheme;
+
+/**
+ * API Controller for the `/roles` resource.
+ */
+class RolesApiController extends AbstractApiController {
+
+    /** @var CapitalCaseScheme */
+    private $caseScheme;
+
+    /** @var Schema */
+    private $idParamSchema;
+
+    /** @var PermissionModel */
+    private $permissionModel;
+
+    /** @var RoleModel */
+    private $roleModel;
+
+    /** @var Schema */
+    private $rolePostSchema;
+
+    /** @var Schema */
+    private $roleSchema;
+
+    /**
+     * RolesApiController constructor.
+     *
+     * @param RoleModel $roleModel
+     */
+    public function __construct(RoleModel $roleModel, PermissionModel $permissionModel) {
+        $this->roleModel = $roleModel;
+        $this->permissionModel = $permissionModel;
+        $this->caseScheme = new CapitalCaseScheme();
+    }
+
+    /**
+     * Delete a role.
+     *
+     * @param int $id The ID of the role.
+     * @return null
+     */
+    public function delete($id) {
+        $this->permission('Garden.Settings.Manage');
+
+        $in = $this->idParamSchema()->setDescription('Delete a role.');
+        $out = $this->schema([], 'out');
+
+        $this->roleByID($id);
+        $this->roleModel->deleteID($id);
+
+        return null;
+    }
+
+    /**
+     * Get a schema instance comprised of all available role fields.
+     *
+     * @return Schema Returns a schema object.
+     */
+    protected function fullSchema() {
+        $schema = Schema::parse([
```

Comment:
We should probably set the $schema variable as static to prevent calling Schema::parse on every call to fullSchema.

Also, that function has been created on every APIController. Maybe we should move it's definition in a parent class.

One solution I'm seeing here is to create an abstract function defineFullSchema() in the parent class. (which would replace this one. Then in the parent constructor do this assignation:
```
$this->fullSchema = $this->defineFullSchema();
```

Every child class will then be able to use $this->fullSchema.
This will also allow to get rid of functions like: `roleSchema()` since we will be able to use `$this->fullSchema` in `schema()` (if I'm not wrong)